Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/CharacterEmoteSystem.cs

```diff
@@ -60,20 +61,37 @@ private void UpdateEmoteTags(ref CharacterEmoteComponent emoteComponent, in IAva
             emoteComponent.CurrentAnimationTag = avatarView.GetAnimatorCurrentStateTag();
         }
 
+        [Query]
+        [None(typeof(CharacterEmoteIntent))]
+        private void CancelEmotesByFlag(ref CharacterEmoteComponent emoteComponent, in IAvatarView avatarView)
```

Comment:
Doing a second query to check the cancellation seems redundant. Since this one contains the same query as `CancelEmotesByTag`, we should consider merging them and then having 2 methods inside the query:
- `CancelEmotesByFlag(ref emoteComponent, in avatarView);`
- `CancelEmotesByTag(ref emoteComponent, in avatarView);`

Reply:
Good point!